synapseclient/models/table_components.py

```diff
@@ -0,0 +1,670 @@
+import os
+from dataclasses import dataclass, field, replace
+from enum import Enum
+from typing import Any, Dict, List, Optional, TypeVar, Union
+
+from typing_extensions import Self
+
+from synapseclient import Column as Synapse_Column
```

Comment:
This python module got all the `dataclass` classes that I originally had in `table_operator.py`